targets/nintendoswitch.s

```diff
@@ -45,18 +44,17 @@ _mod_header:
 .section .text, "x"
 .global start
 start:
-
-    // save lr
-    mov  x7, x30
-
-    // get aslr base
-    bl   +4
-    sub  x6, x30, #0x88
+    // Get ASLR Base
+    adrp x6, _start
 
     // context ptr and main thread handle
     mov  x5, x0
     mov  x4, x1
 
+    // Save lr, context pointer, main thread handler
+    adrp x0, _aslr_base
+    str  x6, [x0, #:lo12:_aslr_base]
+
```

Comment:
I don't understand this. Here you are saving the `x6` variable, and you load it again after clearing the .bss section?

I see that bss zeroing overwrites registers `x5` and `x6`, but doesn't touch `x0` (which you're loading the ASLR base into). So why not move it to `x0` instead?

```suggestion
    mov x0, x6
```

Reply:
Done!